job-runner/chain/chain.go

```diff
@@ -97,103 +135,197 @@ func (c *chain) NextJobs(jobId string) proto.Jobs {
 }
 
 // JobIsReady returns whether or not a job is ready to run. A job is considered
-// ready to run if all of its previous jobs are complete. If any previous jobs
-// are not complete, the job is not ready to run.
+// ready to run if it is Pending or Stopped with some retry attempts remaining,
+// and all of its previous jobs are complete. If any previous jobs are not
+// complete, the job is not ready to run.
 func (c *chain) JobIsReady(jobId string) bool {
```

Comment:
Renaming to `IsRunnable(jobId string) bool` will align with `RunnableJobs()`.

Reply:
If we settle on "runnable" then let's be sure not to use "ready" anywhere, i.e. be consistent with the "runnable job" term and define what that means.